source/geod24/LocalRest.d

```diff
@@ -1176,3 +1233,77 @@ unittest
     node.foo();
     node.bar(1);
 }
+
+// request timeouts (from main thread)
+unittest
+{
+    import core.thread;
+    import std.exception;
+
+    static interface API
+    {
+        size_t sleepFor (long dur);
+    }
+
+    static class Node : API
+    {
+        override size_t sleepFor (long dur)
+        {
+            Thread.sleep(msecs(dur));
+            return 42;
+        }
+    }
+
+    // node with no timeout
+    auto node = RemoteAPI!API.spawn!Node();
+    assert(node.sleepFor(80) == 42);  // no timeout
+
+    // node with a configured timeout
+    auto to_node = RemoteAPI!API.spawn!Node(50.msecs);
+
+    /// none of these should time out
+    assert(to_node.sleepFor(10) == 42);
+    assert(to_node.sleepFor(20) == 42);
+    assert(to_node.sleepFor(30) == 42);
+    assert(to_node.sleepFor(40) == 42);
+
+    auto exc = collectException!Exception(to_node.sleepFor(100));
+    assert(exc !is null);
+}
+
+// request timeouts (foreign node to another node)
+unittest
+{
+    import core.thread;
+    import std.exception;
+
+    static interface API
+    {
+        void check ();
+        size_t sleepFor (long dur);
+    }
+
+    static class Node : API
+    {
+        override void check ()
+        {
+            auto node = RemoteAPI!API.spawn!Node(50.msecs);
```

Comment:
I would not spawn a `Node` from another thread. At the moment `Node`s should be spawned from the main thread.